trestle/tasks/xlsx_to_oscal_component_definition.py

```diff
@@ -0,0 +1,671 @@
+
+    Attributes:
+        name: Name of the task.
+    """
+
+    name = 'xlsx-to-oscal-component-definition'
+
+    def __init__(self, config_object: Optional[configparser.SectionProxy]) -> None:
+        """
+        Initialize trestle task xlsx-to-oscal-component-definition.
+
+        Args:
+            config_object: Config section associated with the task.
+        """
+        super().__init__(config_object)
+        self._timestamp = datetime.datetime.utcnow().replace(microsecond=0).replace(tzinfo=datetime.timezone.utc
+                                                                                    ).isoformat()
+
+    def set_timestamp(self, timestamp) -> None:
+        """Set the timestamp."""
+        self._timestamp = timestamp
+
+    def print_info(self) -> None:
+        """Print the help string."""
+        logger.info(f'Help information for {self.name} task.')
+        logger.info('')
+        logger.info(
+            'Purpose: From spread sheet and catalog produce Open Security Controls Assessment Language (OSCAL) component definition file.'
+        )
+        logger.info('')
+        logger.info('Configuration flags sit under [task.xlsx-to-oscal-component-definition]:')
+        logger.info(
+            '  catalog-file      = (required) the path of the OSCAL catalog file, for example '
+            + self._get_catalog_title() + '.'
+        )
+        logger.info(
+            '  spread-sheet-file = (required) the path of the spread sheet file, containing data for production of component definition.'
+        )
+        logger.info('  work-sheet-name   = (required) the name of the work sheet in the spread sheet file.')
+        logger.info('                      column "a" contains goal ID.')
+        logger.info('                      column "b" contains goal text.')
+        logger.info('                      column "ac-ai" contains controls.')
+        logger.info('                      column "am" contains component name.')
+        logger.info('                      column "an" contains goal name.')
+        logger.info('                      column "ar" contains parameter name and description, separated by newline.')
+        logger.info('                      column "as" contains parameter values.')
+        logger.info(
+            '  output-dir        = (required) the path of the output directory comprising synthesized OSCAL .json files.'
+        )
+        logger.info('  output-overwrite  = (optional) true [default] or false; replace existing output when true.')
+
+    def simulate(self) -> TaskOutcome:
+        """Provide a simulated outcome."""
+        return TaskOutcome('simulated-success')
+
+    def execute(self) -> TaskOutcome:
+        """Provide an executed outcome."""
+        try:
+            return self._execute()
+        except Exception:
+            logger.info(traceback.format_exc())
+            return TaskOutcome('failure')
+
+    def _execute(self) -> TaskOutcome:
```

Comment:
This function is epically long and needs to be shortened / abstracted.

Reply:
Reduced.